templates/prometheus-servicemonitor.yaml

```diff
@@ -0,0 +1,37 @@
+{{ template "vault.mode" . }}
+{{ if .Values.telemetry.prometheusOperator.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "vault.fullname" . }}
+  labels:
+    helm.sh/chart: {{ include "vault.chart" . }}
+    app.kubernetes.io/name: {{ include "vault.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- include "prometheusOperator.selectors" .Values.telemetry.prometheusOperator.serviceMonitor | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "vault.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- if eq .mode "ha" }}
+      vault-active: "true"
+      {{- else }}
+      vault-internal: "true"
+      {{- end }}
+  endpoints:
+  - port: http
```

Comment:
Does this need to match the name of the vault service port? Should it be `{{ include "vault.scheme" . }}` instead? I think it's the same for active and internal:

https://github.com/hashicorp/vault-helm/blob/17bffbcabe0221be0b8e0a02f5a9d8b2b76794f0/templates/server-ha-active-service.yaml#L29-L35

https://github.com/hashicorp/vault-helm/blob/17bffbcabe0221be0b8e0a02f5a9d8b2b76794f0/templates/server-headless-service.yaml#L22-L25

Reply:
Good call! Fixed with be4019d937d74d8d2b1be57b23122ecc1aed9614